packages/puppeteer-core/src/common/Page.ts

```diff
@@ -3113,6 +3122,35 @@ export class CDPPage extends Page {
   ): Promise<HandleFor<Awaited<ReturnType<Func>>>> {
     return this.mainFrame().waitForFunction(pageFunction, options, ...args);
   }
+
+  /**
+   * This method is typically coupled with an action that triggers a device
+   * request from an api such as WebBluetooth.
+   *
+   * :::caution
+   *
+   * This must be called before the device request is made. It will not return a
+   * currently active device prompt.
+   *
+   * :::
+   *
+   * @example
+   *
+   * ```ts
+   * const [deviceRequest] = Promise.all([
+   *   page.waitForDevicePrompt(),
+   *   page.click('#connect-bluetooth'),
+   * ]);
+   * await devicePrompt.select(
+   *   await devicePrompt.waitForDevice(({name}) => name.includes('My Device'))
+   * );
+   * ```
+   */
+  override waitForDevicePrompt(
```

Comment:
can iframes request bluetooth access? Perhaps it makes sense to create deviceRequestPromptManager per frame (or rather per OOP-iframe) so that this method becomes `this.mainFrame().waitForDevicePrompt()`.